src/Microsoft.AspNetCore.StaticFiles/StaticFileMiddleware.cs

```diff
@@ -59,6 +60,7 @@ public StaticFileMiddleware(RequestDelegate next, IHostingEnvironment hostingEnv
             _contentTypeProvider = options.Value.ContentTypeProvider ?? new FileExtensionContentTypeProvider();
             _fileProvider = _options.FileProvider ?? Helpers.ResolveFileProvider(hostingEnv);
             _matchUrl = _options.RequestPath;
+            _responseCacheFilter = new ResponseCacheFilter();
```

Comment:
The only problem with this currently is that the StaticFilesMiddleware doesn't resolve any internal services through DI. All of the existing services in the StaticFileMiddleware are instantiated directly or passed in via the `Options` object. 

I could obviously add the required `IServiceCollection` extensions, but that would require updating all existing apps to add an `AddStaticFiles` call to their `ConfigureServices` call. 

Personally, I would much rather see the whole middleware move to DI but that seems like a big change to make for this (relatively minor) feature? That was the reasoning behind keeping the `IResponseCacheFilter`/`ResponseCacheFilter` internal too - there is currently no way of using a different one.